ops/model.py

```diff
@@ -810,6 +901,291 @@ def __init__(self, name: str, address_info: '_AddressDict'):
         # TODO: expose a hostname/canonical name for the address here, see LP: #1864086.
 
 
+class SecretRotate(enum.Enum):
+    """Secret rotation policies."""
+
+    NEVER = 'never'  # the default in juju
+    HOURLY = 'hourly'
+    DAILY = 'daily'
+    WEEKLY = 'weekly'
+    MONTHLY = 'monthly'
+    QUARTERLY = 'quarterly'
+    YEARLY = 'yearly'
+
+
+class SecretInfo:
```

Comment:
Why do we need a class for `SecretInfo`?  Couldn't we simply call `secret.id` or `secret.revision`? it seems cleaner than `secret.get_info().id`

Reply:
It also separates out "what is visible to Owners" vs "what is visible to Consumers".

Reply:
Yeah, the reason John gives is the main one: most of the items in `SecretInfo` are only available to secret owners.